src/browser/modules/Editor/Monaco.tsx

```diff
@@ -64,6 +64,29 @@ const Monaco = ({
   const editorRef = useRef<editor.IStandaloneCodeEditor | null>(null)
   const monacoId = `monaco-${id}`
 
+  // Share current text with parent and add warnings
```

Comment:
@HerrEmil wanted to get you to just check over this and see if it would still work ok?

Reply:
I checked out and tried and no, these changes have caused an issue. The empty array was to only run the effect once, it creates the editor instance and a listener for text changes. Running that effect on each value change will destroy and recreate the editor instance on each value update. It is noticeable to the user as it will reset the theme and make you lose focus when typing.